### PR TITLE
Add License and CertifyLegal to Arango backend.

### PR DESCRIPTION
**Pull request by jeffmendoza**
# Description of the PR

Add License and CertifyLegal to Arango backend.

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

File: pkg/assembler/backends/arangodb/certifyLegal_test.go [hunk 1]
```diff
@@ -0,0 +1,783 @@
...
+		// 	Calls: []call{
+		// 		{
+		// 			PkgSrc: model.PackageOrSourceInput{
+		// 				Package: p1,
+		// 			},
+		// 			Dec:   []*model.LicenseInputSpec{lNone},
+		// 			Legal: &model.CertifyLegalInputSpec{},
+		// 		},
+		// 	},
+		// 	ExpIngestErr: true,
```
Comment (jeffmendoza) [inline pkg/assembler/backends/arangodb/certifyLegal_test.go:621]: @pxp928 These missing-noun expected errors are hanging the test. Am I doing something wrong in the query?

Comment (pxp928) [reply pkg/assembler/backends/arangodb/certifyLegal_test.go:597]: yes we can remove these as they got moved to the resolvers

**Review by mihaimaruseac — APPROVED** [review]

File: pkg/assembler/backends/arangodb/certifyLegal.go [hunk 2]
```diff
@@ -0,0 +1,882 @@
...
+		aqb.filter("certifyLegal", "attribution", "==", "@attribution")
+		queryValues["attribution"] = *certifyLegalSpec.Attribution
+	}
+	if certifyLegalSpec.Justification != nil {
+		aqb.filter("certifyLegal", justification, "==", "@"+justification)
+		queryValues[justification] = *certifyLegalSpec.Justification
+	}
+	if certifyLegalSpec.TimeScanned != nil {
+		aqb.filter("certifyLegal", "timeScanned", "==", "@timeScanned")
+		queryValues["timeScanned"] = *certifyLegalSpec.TimeScanned
```
Comment (pxp928) [inline pkg/assembler/backends/arangodb/certifyLegal.go:198]: store in UTC? I am updating the others verbs where timestamp is used also

**Review by pxp928 — APPROVED** [review]
LGTM, just one minor comment about the timestamp and using the existing test data

File: pkg/assembler/backends/arangodb/certifyLegal_test.go [hunk 3]
```diff
@@ -0,0 +1,759 @@
...
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/guacsec/guac/internal/testing/ptrfrom"
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+)
+
+var t2 = time.Unix(1e9, 0)
+var t3 = time.Unix(1e9+5, 0)
+
+var p1 = &model.PkgInputSpec{
```
Comment (pxp928) [inline pkg/assembler/backends/arangodb/certifyLegal_test.go:35]: All the testdata can be found here: https://github.com/guacsec/guac/blob/main/internal/testing/testdata/models.go. Arango and Ent both use this in unit tests

File: pkg/assembler/backends/arangodb/license_test.go [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/guacsec/guac/internal/testing/ptrfrom"
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+	"golang.org/x/exp/slices"
+)
+
+var l1 = &model.LicenseInputSpec{
```
Comment (pxp928) [inline pkg/assembler/backends/arangodb/license_test.go:31]: can move this to the https://github.com/guacsec/guac/blob/main/internal/testing/testdata/models.go and use from there so ENT and other backends can also use.

**Review by pxp928 — APPROVED** [review]

File: pkg/assembler/backends/arangodb/certifyLegal_test.go (hunk 1 above)
Comment (pxp928) [reply pkg/assembler/backends/arangodb/certifyLegal_test.go:597]: yeah, the test will timeout (exponential backoff on the query) if the subject is not already ingested. However, the assembler will enforce the ingestion of the nouns before the verbs. Not sure if we need to also test it here.

File: pkg/assembler/backends/arangodb/certifyLegal.go (hunk 2 above)
Comment (jeffmendoza) [reply pkg/assembler/backends/arangodb/certifyLegal.go]: Done.

File: pkg/assembler/backends/arangodb/certifyLegal_test.go (hunk 3 above)
Comment (jeffmendoza) [reply pkg/assembler/backends/arangodb/certifyLegal_test.go]: Done.

File: pkg/assembler/backends/arangodb/license_test.go (hunk 4 above)
Comment (jeffmendoza) [reply pkg/assembler/backends/arangodb/license_test.go]: Done.

File: pkg/assembler/backends/arangodb/certifyLegal_test.go (hunk 1 above)
Comment (jeffmendoza) [reply pkg/assembler/backends/arangodb/certifyLegal_test.go:527]: Sounds good, opened #1360 to track going forward.

**Pull request merged**